projects/packages/connection/src/class-tokens.php

```diff
@@ -357,6 +367,11 @@ public function return_30() {
 	 * @return object|false
 	 */
 	public function get_access_token( $user_id = false, $token_key = false, $suppress_errors = true ) {
+		if ( $this->is_site_locked() ) {
```

Comment:
I think the whole token lock functionality is internal use only, I don't see why any consumer would need it.
So instead of exposing it to the consumers, we should be better off keeping it alongside the rest of token handling.

If the tokens are locked and the domain name doesn't match, the tokens get removed and it looks like they've never been there to the rest of the system.
Do you think other components might use it for their purposes?

Reply:
In light of our discussion I agree that keeping the mechanism here makes sense in addition to renaming it and documenting the logic a bit.